src/ecl/eclWatchTree.ts

```diff
@@ -93,6 +93,16 @@ export class ECLWatchTree extends Tree {
             wuNode.delete();
         });
 
+        vscode.commands.registerCommand("hpccPlatform.protectWU", (wuNode: ECLWUNode) => {
+            wuNode.protect();
+            this.refresh();
+        });
+
+        vscode.commands.registerCommand("hpccPlatform.unprotectWU", (wuNode: ECLWUNode) => {
+            wuNode.unprotect();
+            this.refresh();
```

Comment:
Not sure this refresh is needed?  If it is, it should wait for the "unprotect" to complete?

Reply:
Yes, you need it or the menu doesn't update.

Reply:
Remove `this.refresh` here.